jupyter_telemetry/formatter.py

```diff
@@ -0,0 +1,73 @@
+from pythonjsonlogger import jsonlogger
+
+EVENT_LEVELS = [
+    'unclassified',
+    'confidential',
+    'secret',
+    'top_secret'
+]
+
+EVENT_MAP = {
+    'unclassified': 0,
+    'confidential': 10,
+    'secret': 20,
+    'top_secret': 30
+}
+
+
+class JsonEventFormatter(jsonlogger.JsonFormatter):
+    """Patch the jsonlogger formatter to include levels for telemetry.
+
+    Properties in a logged event that has a level less than
+    the handler's event_level will be dropped from the emitted event.
+    """
+
+    def __init__(self, logger, handler, *args, **kwargs):
+        self.logger = logger
+        self.handler = handler
+
+        # Set the event logging level
+        if hasattr(self.handler, 'event_level'):
+            event_level = self.handler.event_level
+        else:
+            event_level = 'unclassified'
+        self.setEventLevel(event_level)
+
+        super(JsonEventFormatter, self).__init__(*args, **kwargs)
+
+    def setEventLevel(self, event_level):
```

Comment:
Having a public setter API and a protected attribute for the same is not very Pythonic (much of the stdlib logging module is this way for historical reasons, but it doesn't need to be added to). This can be accomplished with a single property, eliminating the `setEventLevel` setter:

```python
@property
def event_level(self):
    return self._event_level

@event_level.setter
def event_level(self, event_level):
    if event_level not in EVENT_LEVELS:
         raise Exception("Event level '{}' not understood.".format(event_level))
    self._event_level = self.handler.event_level
```

Reply:
Yes, that is true— but in this case, we're overriding the method from the `JsonFormatter` parent class (which is an external dependency). This method is called by other places in this parent class. I chose to keep this method signature consistent with the parent class. I realize this isn't Pythonic, but it's consistent with the method it overrides.

We can, of course, change how this method to use the public setter API... i.e. use the **same method signature** but decorate it with `.setter`? Then, we can have a protected `event_level` attribute. Is that what you propose?

Reply:
Looks like this was recently changed! Ignore my prior comment. :)

Updating to use property setter.